### ISSUE TITLE
Model overfitted or mapping challenge dataset not diverse? Poor performance with custom images

### ISSUE THREAD

**Issue by carbonox-infernox** [neptune-ml/open-solution-mapping-challenge#196]
Excellent results can be achieved using the Developers' fully-trained model (found here: https://app.neptune.ml/neptune-ml/Mapping-Challenge/files) to predict on the test_images set provided as part of the mapping challenge. However, over the past few days I have experimented with custom data from Bing imagery and Hexagon and have gotten very poor results. Here are some examples:

![subdivision_10_18](https://user-images.githubusercontent.com/27938083/47742991-4b179380-dc54-11e8-802a-d98943cdd36e.jpg)
![subdivision_03_22](https://user-images.githubusercontent.com/27938083/47743000-54086500-dc54-11e8-84d7-ceda52ec7174.jpg)
![subdivision_04_02](https://user-images.githubusercontent.com/27938083/47743019-5f5b9080-dc54-11e8-8193-dd62731b5bf8.jpg)
![subdivision_17_11](https://user-images.githubusercontent.com/27938083/47743027-6387ae00-dc54-11e8-9c5c-92b0986f9c9f.jpg)
![subdivision_03_03](https://user-images.githubusercontent.com/27938083/47743054-726e6080-dc54-11e8-8a29-c02ef41e4058.jpg)
(a few were good)
![subdivision_01_09](https://user-images.githubusercontent.com/27938083/47743518-a302ca00-dc55-11e8-8c02-c01c8cb56d9f.jpg)

Where cyan circles denote mask centroids. I have experimented with several factors, such as:

- image source (bing vs hexagon)
- image quality (high vs low areas)
- zoom level (pixels per meter given fixed 300x300 image size)
- quality level due to conversion from png to jpg
- overlapping tiles and keeping only masks found in both

And I've treated my custom data exactly as I treated test data. I used the same pipeline and settings for both. I'm sort of forced to conclude that either the model has overfitted to the "type" of data given by the open challenge, or said given data is not a diverse selection of image sources.

Has anyone else found good results with custom imagery? If so, what was the source?

**Comment by jakubczakon** [comment]
@carbonox-infernox good job analyzing the results on external data.
I do think that the dataset provided by the organizers was not very diverse and was provided more as a proof of concept than anything else.

So in short, I think the model just learned what it was fed. A simple building dataset.
That being said I think (talked about it with other participants), that you can finetune the model we have for just a few epochs on a new dataset successfully.

I'd suggest you try that.

**Comment by 51616** [comment]
@carbonox-infernox Can you guide us through how you load the model weights and get prediction on your own images? Along with code examples would make this very clear. :)

**Comment by apyskir** [comment]
Hi @51616 Just download the model and place it in your experiment directory. In `neptune.yaml` you have to specify `experiment_dir`. Inside this directory you should create a `models` directory and place your `unet` file there. While inferencing the code will look for a file with model weights in that directory. Hope it helps.
ps.2. It's prepared for more than one class, so the first channel refers to "background" class, and the other refers to "building" class. (I hope I remember this correctly)

**Comment by Angel0003** [comment]
Can you guide us through how to train my own datasets? Along with preparing training data and code examples would make this very clear. :),how to generate the file of metadata.csv?
